submissions/5mountains/tiny-js-world-oop/index.js

```diff
@@ -0,0 +1,105 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+class Resident {
+  constructor(species, name, gender, saying, friends) {
+    this.species = species;
+    this.name = name;
+    this.gender = gender;
+    this.saying = saying;
+    this.friends = friends;
+  }
+  introducing() {
+    return `Wonderful creature - <b>${
+      this.species
+    }</b>, whose name is <b>${
+      this.name
+    }</b>! Usually the greeting is <b>${
+      this.saying
+    }</b></i> and friends of this inhabitant are <b>${
+      this.friends.join(', ')
+    }</b>. By the wat, this member is owner of`;
```

Comment:
It is not a really good practice for a superclass to assume what child classes would do (add something about their limbs in this case).
Imagine you have a really vast hierarchy of class including limbless creatures like worms. It wouldn't be wise to force every child class to be an owner of anything.

(Yet we assume our inhabitans may have 0 or more friends).

Reply:
Aha, so the best idea that superclass contains only a few major features as name, gender, and species. And other characteristics should be added with the constructor in each extended class?)

Reply:
@5Mountains exactly